controllers/packetmachine_controller.go

```diff
@@ -248,11 +252,15 @@ func (r *PacketMachineReconciler) reconcileDelete(ctx context.Context, machineSc
 
 	device, err := r.PacketClient.GetDevice(providerID)
 	if err != nil {
+		if err.(*packngo.ErrorResponse).Response != nil && err.(*packngo.ErrorResponse).Response.StatusCode == http.StatusNotFound {
+			// When the server does not exist we do not have anything left to do.
+			// Probably somebody manually deleted the server from the UI or via API.
+			logger.Info("Server not found, nothig left to do")
+			controllerutil.RemoveFinalizer(packetmachine, infrastructurev1alpha3.MachineFinalizer)
+			return ctrl.Result{}, nil
+		}
 		return ctrl.Result{}, fmt.Errorf("error retrieving machine status %s: %v", packetmachine.Name, err)
 	}
-	if device == nil {
```

Comment:
We should not remove this. It is a good safety check. Instead, we should add the `RemoveFinalizer` here as well.

Reply:
fixed